### PR TITLE
setup: finish front setup

### PR DESCRIPTION
**Pull request by BarnabePILLIAUDIN**
# Done

## Frontend

- Setup Next.JS with app router
- Setup eslint + prettier
- Setup NextUI
-Setup script for check format + fix format + running check of linter and format (can be used in CI later)

### CODE REVIEW

**Review by cedricgautier — COMMENTED** [review]
Please remove .idea folder

File: front/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "format:check": "prettier -c .",
+    "format:fix": "prettier -w .",
+    "ci-check": "bun run format:check && bun run lint"
+  },
+  "dependencies": {
+    "@nextui-org/react": "^2.4.2",
+    "eslint-config-prettier": "^9.1.0",
+    "framer-motion": "^11.2.12",
+    "next": "14.2.4",
+    "prettier": "^3.3.2",
```
Comment (cedricgautier) [inline front/package.json:19]: I think this is a dev dependency

File: front/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
+    "format:check": "prettier -c .",
+    "format:fix": "prettier -w .",
+    "ci-check": "bun run format:check && bun run lint"
+  },
+  "dependencies": {
+    "@nextui-org/react": "^2.4.2",
+    "eslint-config-prettier": "^9.1.0",
```
Comment (cedricgautier) [inline front/package.json:16]: same here, I think this is a dev dependency

File: front/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "front",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint",
+    "format:check": "prettier -c .",
```
Comment (cedricgautier) [inline front/package.json:10]:
```suggestion
    "format-check": "prettier -c .",
```

File: front/package.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    "format:fix": "prettier -w .",
```
Comment (cedricgautier) [inline front/package.json:11]:
```suggestion
    "format-fix": "prettier -w .",
```

File: front/tailwind.config.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+/** @type {import('tailwindcss').Config} */
+import { nextui } from "@nextui-org/react"
+
+module.exports = {
+  content: [
+    "./src/pages/**/*.{js,ts,jsx,tsx,mdx}",
```
Comment (cedricgautier) [inline front/tailwind.config.js:6]: We're not using the page router, you can remove this I think

File: front/tailwind.config.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    "./src/app/**/*.{js,ts,jsx,tsx,mdx}",
+    "./node_modules/@nextui-org/theme/dist/**/*.{js,ts,jsx,tsx}"
+  ],
+  theme: {
+    extend: {
+      backgroundImage: {
+        "gradient-radial": "radial-gradient(var(--tw-gradient-stops))",
+        "gradient-conic": "conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))"
+      }
+    }
```
Comment (cedricgautier) [inline front/tailwind.config.js:17]: I don't think we need to keep the background images

File: front/tailwind.config.js (hunk 5 above)
Comment (BarnabePILLIAUDIN) [reply front/tailwind.config.js]: Done

File: front/tailwind.config.js (hunk 6 above)
Comment (BarnabePILLIAUDIN) [reply front/tailwind.config.js:16]: Required for NextUI

Comment (cedricgautier) [reply front/tailwind.config.js:16]: Ah ok, thx

**Review by cedricgautier — APPROVED** [review]

**Review by Gougoh — APPROVED** [review]

**Pull request merged**